src/Plugin/EntityReferenceSelection/OgSelection.php

```diff
@@ -77,37 +139,45 @@ protected function buildEntityQuery($match = NULL, $match_operator = 'CONTAINS')
       $query->condition($bundle_key, $bundles, 'IN');
     }
 
-    $user_groups = $this->getUserGroups();
-    if (!$user_groups) {
+    // Get the identifier key of the entity.
+    $identifier_key = $definition->getKey('id');
+
+    if ($this->currentUser->isAnonymous()) {
+      // @todo: Check if anonymous users should have access to any referenced
+      // groups? What about groups that allow anonymous posts?
+      return $query->condition($identifier_key, -1);
+    }
+
+    if ($this->currentUser->hasPermission('administer group')) {
+      // User can see all the groups.
       return $query;
     }
 
-    $identifier_key = $definition->getKey('id');
+    if (empty($this->configuration['entity'])) {
+      // @todo: Find out why we have this scenario.
```

Comment:
That part was added in https://github.com/Gizra/og/pull/181/commits/8c940139238a47f90df722d9888b5a5666b507f3 which committed at Oct 15, 2016. The patch for core came at (Nov 14, 2016)[https://github.com/Gizra/og/pull/181#issuecomment-260302310] so I don't think the commit relate to the patch.

Anyway, maybe we can tackle it in #256?